docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/upgrade-guides/6-0-0-upgrade-guide/index.md

```diff
@@ -0,0 +1,153 @@
+```json
+{
+   // 1-0-0
+   "properties": {
+      "a": {"type": "integer"}
+   }
+}
+```
+
+```json
+{
+   // 1-0-1
+   "properties": {
+      "a": {"type": "string"}
+   }
+}
+```
+
+Because `1-0-1` events cannot be loaded into the same table with `1-0-0`, the data would be put in a separate table, e.g. `com_snowplowanalytics_ad_click_1_0_1_recovered_9999999`, where:
+  - `1_0_1` is the version of the offending schema;
+  - `9999999` is a hash code unique to the schema (i.e. it will change if the schema is overwritten with a different one).
+
+If you create a new schema `1-0-2` that reverts the offending changes and is again compatible with `1-0-0`, the data for events with that schema will be written to the original table as expected.
+
+### Identification of the schemas that need patching
+
+After the rdb-loader's upgrade, users might find out that some old schema lineages land in the recovery table.
+
+To avoid this, older iglu schemas must be patched to align with the latest.
+
+To help with the identification of the schemas that need patching, we have rewritten the rdbms table-check command.
+`igluctl` will now cross-check the types between the iglu server and RedShift or Postgres. Depending on what types
+offended the schema errors might vary.
+
+Procedure of identifying offending warehouse patches is as follows:
+
+1) Run `igluctl static generate` command. If recovery table to be created it will show up as a warning message. Example:
+```bash
+mkdir <schemas_folder> <sql_folder>
+igluctl static pull <schemas_folder> <iglu_url> <iglu_key>
+igluctl static generate <schemas_folder> <sql_folder> 
+# ...
+# iglu:com.snowplowanalytics.iglu/resolver-config/jsonschema/1-0-3 has a breaking change Incompatible encoding in column cache_size old type RedshiftBigInt/ZstdEncoding new type RedshiftDouble/RawEncoding
+# ...
+```
+
+2) Run `table-check` to crosscheck if warehouse was patched. This is done in case if  schema lineage had a broken schema
+   that were supreseed and no longer used, for example:
+```
+100 - {"type" : "integer"}
+101 - {"type" : "number"}  // deprecated
+102 - {"type" : "integer"} // new corrected version
+```
+Situation above is not an issue, but would produce a warning in step 1).
+```bash
+igluctl --server <iglu_url> \
+        --apikey <iglu_key> \
+        --host <redshift_host> \
+        --port <redshift_port> \
+        --username <username> \
+        --password <password> \
+        --dbname  <database> \
+        --dbschema <schema>
+# ...
+# * Comment problem - SchemaKey found in table comment [iglu:com.test/test/jsonschema/1-0-0] does not match expected [iglu:com.test/test/jsonschema/1-0-1]
```

Comment:
I don’t fully understand what this means. Why would the table comment be 1-0-0 when a newer schema exists? Did the user alter the table manually? I think I am missing the link between step 1 and this case in step 2. In step 1 we said “check if you have schema versions that will create recovery tables”. Then we said step 2 would be to see which of those we need to worry about (the above example shows that we don’t need to worry about schemas that were never used... btw is that the case in Redshift? Doesn’t it always try to migrate the table as soon as the schema is created?). But now we are somehow discovering that tables were manually altered?